src/coordinator.rs

```diff
@@ -624,6 +626,12 @@ impl Coordinator {
         if !matches.is_empty() {
             tracing::info!(?matches, "Matches found");
         }
+        
+        // Sort the matches by distance in ascending order
+        matches.sort_by(|a, b| a.distance.partial_cmp(&b.distance).unwrap());
```

Comment:
You can use [sort_by_key](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_by_key)

Reply:
Hmm, it seems it's not implemented for `f64` type. Please see [here](https://github.com/rust-lang/rust/issues/67417).

Reply:
Ah right, makes sense. Although instead of `partial_cmp` your might want to use [f64::total_cmp](https://doc.rust-lang.org/std/primitive.f64.html#method.total_cmp)